samples/msgext-unfurling-ac-loop-components/nodejs/bots/teamsBot.js

```diff
@@ -0,0 +1,143 @@
+const axios = require("axios");
+const querystring = require("querystring");
+const { TeamsActivityHandler, CardFactory } = require("botbuilder");
+
+class TeamsBot extends TeamsActivityHandler {
+  constructor() {
+    super();
+  }
+
+  // Msgext-link-unfurling
+  handleTeamsAppBasedLinkQuery() {
+    const card = CardFactory.adaptiveCard({
+        "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+        "type": "AdaptiveCard",
+        "version": "1.6",
+        "metadata": {
+            "webUrl": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main"
+        },
+        "body": [
+            {
+            "type": "TextBlock",
+            "text": "Adaptive Card-based Loop component",
+            "size": "Large",
+            "weight": "Bolder"
+            },
+            {
+            "type": "Container",
+            "items": [
+                {
+                "type": "TextBlock",
+                "text": "These samples are designed to help understand Microsoft Teams platform capabilities and scenarios(Tabs,Bots,",
+                "weight": "bolder",
+                "size": "medium"
+                }
+            ]
+         }
+        ]
+    });
+
+    const attachment = card;
+
+    attachment.preview = {
+        content: {
+            title: "Thumbnail Card",
+            text:"Test",
+        },
+        contentType: "application/vnd.microsoft.card.thumbnail",
+    }
+
+    const result = {
+        attachmentLayout: 'list',
+        type: 'result',
+        attachments: [attachment]
+    };
+
+    const response = {
+        composeExtension: result
+    };
+
+    return response;
+}
+
+  // Message extension Code Search
+  // Used in creating a Search-based Message Extension
+  async handleTeamsMessagingExtensionQuery(context, query) {
```

Comment:
@Harikrishnan-MSFT The implementation inside this method and other two are completely unrelated, can we bring some coherence here?

Reply:
Fixed